std/path.d

```diff
@@ -1736,7 +1736,7 @@ if (isSomeChar!C)
         if (result)
             result = chainPath(result, path).array;
         else
-            result = path;
+            result = path.idup;
```

Comment:
Thinking about it: wouldn't `assumeUnique` be better here to avoid an unnecessary allocation?
I know the loop is already horrible with `.array` on every iteration, but that's not an argument to make it even worse?
To ensure that the new content is freshly allocated, we could check this after the loop is then and only `idup` then...